contribute/style-guides/testing.md

```diff
@@ -0,0 +1,251 @@
+
+- [screen.debug()](https://testing-library.com/docs/queries/about/#screendebug) - This function prints a human-readable representation of the document's DOM tree when called without arguments, or the DOM tree of specific node(s) when provided with arguments. It is internally using `console.log` to log the output to terminal.
+- [Testing Playground](https://testing-playground.com/) - An interactive sandbox that allows testing which queries work with specific HTML elements.
+- [logRoles](https://testing-library.com/docs/dom-testing-library/api-debugging/#prettydom) - A utility function that prints out all the implicit ARIA roles for a given DOM tree.
+
+## Testing Select Components
+
+Here, the [OrgRolePicker](https://github.com/grafana/grafana/blob/main/public/app/features/admin/OrgRolePicker.tsx) component is used as an example. This component essentially serves as a wrapper for the `Select` component, complete with its own set of options.
+
+```tsx
+import { OrgRole } from '@grafana/data';
+import { Select } from '@grafana/ui';
+
+interface Props {
+  value: OrgRole;
+  disabled?: boolean;
+  'aria-label'?: string;
+  inputId?: string;
+  onChange: (role: OrgRole) => void;
+  autoFocus?: boolean;
+  width?: number | 'auto';
+}
+
+const options = Object.keys(OrgRole).map((key) => ({ label: key, value: key }));
+
+export function OrgRolePicker({ value, onChange, 'aria-label': ariaLabel, inputId, autoFocus, ...restProps }: Props) {
+  return (
+    <Select
+      inputId={inputId}
+      value={value}
+      options={options}
+      onChange={(val) => onChange(val.value as OrgRole)}
+      placeholder="Choose role..."
+      aria-label={ariaLabel}
+      autoFocus={autoFocus}
+      {...restProps}
+    />
+  );
+}
+```
+
+### Querying the Select Component
+
+There are a few methods to query the `Select` component:
+
+1. One approach is to explicitly pass the `aria-label` prop and then use the `getByRole` method:
+
+```tsx
+describe('OrgRolePicker', () => {
+  it('should render the picker', () => {
+    setup(<OrgRolePicker value={OrgRole.Admin} aria-label={'Role picker'} onChange={() => {}} />);
+    expect(screen.getByRole('combobox', { name: 'Role picker' })).toBeInTheDocument();
+  });
+});
+```
+
+2. Alternatively, add a `label` element and provide the `htmlFor` prop. In this scenario, a matching `inputId` should be passed to the `Select` component:
+
+```tsx
+describe('OrgRolePicker', () => {
+  it('should render the picker', () => {
+    setup(
+      <>
+        <label htmlFor={'role-picker'}>Role picker</label>
+        <OrgRolePicker value={OrgRole.Admin} inputId={'role-picker'} onChange={() => {}} />
+      </>
+    );
+    expect(screen.getByRole('combobox', { name: 'Role picker' })).toBeInTheDocument();
+  });
+});
+```
+
+It's also possible to use `*ByLabelText` queries. However, the `*ByRole` queries are [more robust](https://testing-library.com/docs/queries/bylabeltext/#name) and are generally recommended over the former.
```

Comment:
I think this is a good shout out maybe even worth pulling up higher into like a 'best practices' section or something?